modin/engines/base/frame/data.py

```diff
@@ -973,7 +973,7 @@ def merge(left_index, right_index):
                 return left_index.join(right_index, how=how, sort=sort)
 
         # define condition for joining indexes
-        all_indices_equal = all(indexes[0].equals(index) for index in [indexes[1:]])
+        all_indices_equal = all(indexes[0].equals(index) for index in indexes[1:])
```

Comment:
I'm not sure I understand what these changes do - before were we making a 2D list with one row that contains indices, but now we're just looping through the list of indices?

Reply:
Yes, it was a bug. Deleting these two characters fixes the case in the issue.